pex/resolver.py

```diff
@@ -733,11 +733,13 @@ def resolve(
       prerelease or development versions will override this setting.
     :keyword bool transitive: Whether to resolve transitive dependencies of requirements.
       Defaults to ``True``.
-    :keyword interpreter: The interpreter to use for building distributions and for testing
-      distribution compatibility. Defaults to the current interpreter.
+    :keyword interpreter: If specified, distributions will be resolved for this interpreter, and
+      non-wheel distributions will be built against this interpreter. If both `interpreter` and
+      `platform` are ``None`` (the default), this defaults to the current interpreter.
     :type interpreter: :class:`pex.interpreter.PythonInterpreter`
-    :keyword str platform: The exact target platform to resolve distributions for. If ``None`` or
-      ``'current'``, resolve for distributions appropriate for `interpreter`.
+    :keyword str platform: The exact PEP425-compatible platform string to resolve distributions for,
+      in addition to the platform of the given interpreter, if provided. The current interpreter
```

Comment:
> The current interpreter will be used to build any non-wheel distributions.

This is not true. If non-wheel distributions are encountered when resolving for the given platform, resolution will fail.

Reply:
Perhaps the confusion is that the CLI, which uses the API, will turn `--platform ...` arguments into `PythonInterpreters` when -`--resolve-local-platforms`. It then passes interpreters for those platforms it could resolve instead of passing platforms. Does that clear things up? I.E.: `--resolve-local-platforms` is a CLI feature and not a resolve API feature.

Reply:
What if the `platform` matches the `supported_tags` of the current interpreter? I'm looking at https://github.com/pantsbuild/pex/blob/e67412c31648f6e0d0f2a68b9d38dffabb258ebc/pex/distribution_target.py#L31-L34 and https://github.com/pantsbuild/pex/blob/e67412c31648f6e0d0f2a68b9d38dffabb258ebc/pex/interpreter.py#L61-L72, and it seems like `tags.sys_tags()` can return more than one element in its tuple: https://github.com/pantsbuild/pex/blob/e67412c31648f6e0d0f2a68b9d38dffabb258ebc/pex/vendor/_vendored/packaging/packaging/tags.py#L372-L374

Reply:
This code here:
https://github.com/pantsbuild/pex/blob/cff40893dd1b0ae185201f4871baae87a9fe1335/pex/bin/pex.py#L758-L786

Reply:
> What if the platform matches the supported_tags of the current interpreter? I'm looking at

We raced each other. I think you're confusing a CLI feature for an API feature. Let me know if I've guessed your confusion or if it persists.

Reply:
I was referring to the fact that `DistributionTarget.is_foreign` is calculated according to the tags of the current interpreter (`PythonInterpreter.get()`) when its `._interpreter` field is None (when it's created via `.for_platform()`). While it's true that it would be quite unlikely for `.is_foreign` to return False, if the `platform` provided matches the current interpreter's tags, then, it would actually use the current interpreter to build any non-wheels.

If we actually wanted to document this rare edge case, I might think something like:
> If the platform matches the current interpreter, the current interpreter will be used to build any non-wheels.

could be reasonable.

Reply:
Which isn't exactly rare, now that I think of it -- I believe `platform='current'` would do the trick. I think it would be useful to add the sentence I proposed above, but if that seems like too much information I'm not beholden to it.

Reply:
Ah! But we have a specific check for that! https://github.com/pantsbuild/pex/blob/e67412c31648f6e0d0f2a68b9d38dffabb258ebc/pex/resolver.py#L972-L974

It seems like we might want to consider expanding that check to cover when the platform *matches* any of the given interpreters, in addition to checking if it's just `None`. I don't know if there's any deduplication later. I'll test out what happens if you give an explicit platform string matching the current interpreter.

Reply:
You're losing me. The block of code you point to just ensures that for `platforms=[None]` (which is how `parsed_platforms` `translates platforms=['current']`) and `interpreters=None` or `interpreters=[]` we use  the current interpreter as the single interpreter to resolve with.

The general search you describe is what the CLI does in the code I pointed to above in pex/bin/pex.py when you sepcify `--resolve-local-platforms`.

Reply:
If you want to document the edge case where `platforms=['X', ...]` and 'X' happens to match the current interpreter accessing the Pex API, that's fine with me. But, to be clear, 'X' could either be 'current' in that case or a full platform string that just happens to match the current interpreter.

Reply:
Yes, thank you for piecing that together. Let me know if 2fc5078b09e24ff82ac8eacd984e51a082d1a5d7 seems sufficient to document that.